Don't enqueue Gutenberg styles if Gutenberg isn't running

pento opened the pull request:
#2951 introduced a bug where Gutenberg scripts and styles were being enqueued on every page, instead of just the editor page.

pento committed:
Don't enqueue Gutenberg styles if Gutenberg isn't running on this page

youknowriad reviewed (approved):
Fixes the issues for me!

I wonder if it's not possible to just call `gutenberg_editor_scripts_and_styles` in `gutenberg_init` or something instead of having two entry points.

pento committed:
Switch to just adding the action...

...once we've decided to load Gutenberg, instead of having a kind of weird side-effect function.

pento commented:
@youknowriad: That's a fair point, the weird side effect function was kind of bugging me, too.

The new commit is nicer, I think.

youknowriad commented:
@pento I like it 👍
